ktor-client/ktor-client-features/ktor-client-json/src/io/ktor/client/features/json/JsonFeature.kt

```diff
@@ -48,21 +56,34 @@ class JsonFeature(val serializer: JsonSerializer) {
             scope.requestPipeline.intercept(HttpRequestPipeline.Transform) { payload ->
 
                 context.accept(ContentType.Application.Json)
-                if (context.contentType()?.match(ContentType.Application.Json) != true) {
+                // Only transform if payload is JsonContent.. ie. let manual json serialized pass through
+                if (context.contentType()?.match(ContentType.Application.Json) != true || payload !is JsonContent<*>) {
                     return@intercept
                 }
 
                 context.headers.remove(HttpHeaders.ContentType)
-                proceedWith(feature.serializer.write(payload))
+                proceedWith(feature.serializer.write(payload.value))
             }
 
             scope.responsePipeline.intercept(HttpResponsePipeline.Transform) { (info, response) ->
-                if (response !is HttpResponse
-                    || context.response.contentType()?.match(ContentType.Application.Json) != true
-                ) return@intercept
-
-                proceedWith(HttpResponseContainer(info, feature.serializer.read(info, response)))
+                if (context.response.contentType()?.match(ContentType.Application.Json) == true){
+                    val responseToParse = when (response){
+                        is HttpResponse -> response
+                        is String -> ResponseWithContent(context.response, response) // Undo HttpPlainText feature
+                        else -> return@intercept
+                    }
+                    val unwrappedInfo = unwrapJsonContent(info)
```

Comment:
Could you add test for not `JsonContent` response in info? 
Should we really fail here?

Reply:
No. It seems more symmetric to pass on it here also.